src/acl/external/AD_group/ext_ad_group_acl.cc

```diff
@@ -631,12 +635,12 @@ Valid_Global_Groups(char *UserName, const char **Groups)
     char User[DNLEN + UNLEN + 2];
     size_t j;
 
-    wchar_t *User_DN, *User_LDAP_path, *User_PrimaryGroup;
-    wchar_t **wszGroups, **tmp;
+    wchar_t *User_DN = nullptr, *User_LDAP_path = nullptr;
+    wchar_t *User_PrimaryGroup = nullptr;
     IADs *pUser;
     HRESULT hr;
 
-    strncpy(NTDomain, UserName, sizeof(NTDomain));
+    strncpy(NTDomain, UserName, sizeof(NTDomain)-1);
```

Comment:
Please use our provided `xstr*()` function which avoids the worst truncation issues:
```suggestion
    xstrncpy(NTDomain, UserName, sizeof(NTDomain)-1);
```

There is at least one other (`strncpy(User, NTDomain, DNLEN + UNLEN + 2);`) call which has exactly the same issues as this one does. Please fix that as well.

Reply:
done in commit 9f10dc0de9